### ISSUE TITLE
gui: Searching for any string out of ASCII locks the gui

### ISSUE THREAD

**Issue by 12345ieee** [scanmem/scanmem#173]
Steps to reproduce:
Type: String
Value: Anything outside of ASCII, for example: "à" or "☺☺☺☺"
Hit search.

Stacktrace:

```
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "GameConqueror.py", line 926, in scan_thread_func
    self.backend.send_command(cmd)
  File "/home/andreas/Progetti/scanmem/gui/backend.py", line 68, in send_command
    self.lib.backend_exec_cmd(ctypes.c_char_p(cmd.encode()))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 2: ordinal not in range(128)
```

**Comment by sriemer** [comment]
@12345ieee: Do you have a fix for this?

**Comment by 12345ieee** [comment]
It's part of a muuuuuch bigger issue, that is the unicode-bytes conversion differences between python 2/3.

On top of this, I have no clue how scanmem handles non-ascii strings, you ought to document the backend somehow (aka `libscanmem`) if you want to offload front-end development.

IMHO we'll need a better solution for the python 2/3 double support, either going through `python-six/future` or dropping python2.

**Comment by sriemer** [comment]
Python 2 still works best. The unicode conversion done for Python 3 is still not exactly the same like Python 2 does. E.g. a "\n" string in the hexview is a good example for this.

**Comment by 12345ieee** [comment]
While I agree, dropping py3 would be nonsensical.
We'll talk about this for v0.17, together with #195 .

**Comment by 12345ieee** [comment]
I fixed both this and #172 in ddafed5f824439ca4272de76efbaef46f18ea800 .
It hasn't been fun.

I left your `u()` function in the hexview, replaced all the rest.
Everything I tested (e.g. searching for `èàò\\\èòò\\\` in y2&3) works well.

**Comment by sriemer** [comment]
Merged to master branch. So closing.
